common/openssl-compat.h

```diff
@@ -10,8 +10,11 @@
 #ifndef LIBCRYPTO_COMPAT_H
 #define LIBCRYPTO_COMPAT_H
 
+extern char openssl_compat_used;
```

Comment:
is this needed? I can't see that we use this global anywhere (and if we do, it should probably be const?)

Reply:
It's there to avoid an error if we have no symbols at all. And I just made it pseudo-useful while at it..

Reply:
I guess we want that to not have complex logic for when to include it in the cmake files? 

it seems like others do things like: `extern int make_iso_compilers_happy;` in the empty case (of the c file that is) to make the compiler happy without adding a symbol to all resulting objects.

Reply:
Fixed in f844cb1a5504b05ae7d9ee4af03d85a8d3558162

Reply:
As to why: We need openssl-compat for LibreSSL as well, so logic here would have been too complex. Also, we do it this way in yubico-piv-tool. And we already had conditional compilation in the source code so not really needed.